CMakeLists.txt

```diff
@@ -160,18 +168,46 @@ SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 SET(CMAKE_INSTALL_RPATH "$ORIGIN:$ORIGIN/lib:$ORIGIN/lib/plugins:$ORIGIN/../lib:$ORIGIN/../lib/plugins")
 
 SET(BLA_STATIC ON)
+if(NOT DEFINED ZLIB_USE_STATIC_LIBS)
+  SET(ZLIB_USE_STATIC_LIBS TRUE)
+endif()
 if(NOT DEFINED OPENSSL_USE_STATIC_LIBS)
   SET(OPENSSL_USE_STATIC_LIBS TRUE)
 endif()
 
-if (CUDAQ_BUILD_SELFCONTAINED)
-  # Using -static-libgcc -static-libstdc++ offers finer control here if needed.
+if(NOT DEFINED CUDAQ_ENABLE_STATIC_LINKING)
+  SET(CUDAQ_ENABLE_STATIC_LINKING $ENV{CUDAQ_ENABLE_STATIC_LINKING})
+endif()
+if (CUDAQ_ENABLE_STATIC_LINKING)
+  # Using -static-libgcc -static-libstdc++ offers more fine grained control here if needed.
   # At the time of writing, there wasn't any benefit to using those over -static.
+  # However, while this so far seems to be working reasonably well, this will
+  # unfortunately also link glibc statically, which in general is likely to cause
```

Comment:
Is this true? I'm sort of surprised that CMake would go through the hoops here, and do something that is arguably really dangerous/broken. https://news.ycombinator.com/item?id=23816748#:~:text=Yep.,excuses%20that%20make%20it%20impossible.

Reply:
I am not sure it is per se dangerous, but certainly won't work well in all cases, and it does produce an explicit warning that glibc still needs to be available (separately) at runtime. I think that a lot of the problems are primarily for building libraries that are then potentially later linked with other things, and maybe this configuration can work well to create a fully self-contained executable in some cases.

Reply:
Here is a good answer to why one doesn't want to statically link glibc. "Dangerous" might not be the best adjective here, but it was used as a short hand for programs that can run fine, with weird, unexpected behavior, or arbitrary failures. Which is not ideal. 💥 

https://stackoverflow.com/questions/57476533/why-is-statically-linking-glibc-discouraged

Reply:
Agreed - I've come across and read the same thread. To be revised in a subsequent change.